src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -389,8 +389,10 @@ static IEnumerable<Instruction> CompileBindingPath(ElementNode node, ILContext c
 				n = n.Parent as IElementNode;
 			}
 
-			if (dataTypeNode is null)
+			if (dataTypeNode is null) {
+				context.LoggingHelper.LogWarningOrError(10101, context.XamlFilePath, node.LineNumber, node.LinePosition, 0, 0, $"Binding could be compiled if x:DataType is specified", null);
 				yield break;
+			}
 
 			if (dataTypeNode is ElementNode enode
```

Comment:
Should this case also produce the warning (or some version of it)?

Reply:
This is the case when the user specify x:DataType="{x:Null}" explicitly. so yeah, why not